src/transformers/pipelines/object_detection.py

```diff
@@ -0,0 +1,173 @@
+            if image.startswith("http://") or image.startswith("https://"):
+                # We need to actually check for a real protocol, otherwise it's impossible to use a local file
+                # like http_huggingface_co.png
+                return Image.open(requests.get(image, stream=True).raw)
+            elif os.path.isfile(image):
+                return Image.open(image)
+        elif isinstance(image, Image.Image):
+            return image
+
+        raise ValueError(
+            "Incorrect format used for image. Should be an url linking to an image, a local path, or a PIL image."
+        )
+
+    def __call__(
+        self,
+        images: Union[str, List[str], "Image", List["Image"]],
+        threshold: Optional[float] = 0.9,
+    ) -> Union[Predictions, List[Prediction]]:
+        """
+        Detect objects (bounding boxes & classes) in the image(s) passed as inputs.
+
+        Args:
+            images (:obj:`str`, :obj:`List[str]`, :obj:`PIL.Image` or :obj:`List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images, which must then be passed as a string.
+                Images in a batch must all be in the same format: all as http links, all as local paths, or all as PIL
+                images.
+            threshold (:obj:`float`, `optional`, defaults to 0.9):
+                The probability necessary to make a prediction.
+
+        Return:
+            A list of dictionaries or a list of of list of dictionaries containing result. If the input is a single
+            image, will return a list of dictionaries, if the input is a list of several images, will return a list of
+            list of dictionaries corresponding to each image.
+
+            The dictionaries contain the following keys:
+
+            - **label** (:obj:`str`) -- The label identified by the model. There could be duplicates within the list
+              (corresponding to object of same label within an image).
+            - **score** (:obj:`int`) -- The score attributed by the model for that label.
+            - **box** (:obj:`List[Dict[str, int]]`) -- The bounding box of detected object in image's original size.
+        """
+        is_batched = isinstance(images, list)
+
+        if not is_batched:
+            images = [images]
+
+        images = [self.load_image(image) for image in images]
+
+        with torch.no_grad():
+            inputs = self.feature_extractor(images=images, return_tensors="pt")
+            outputs = self.model(**inputs)
+
+            target_sizes = torch.IntTensor([[im.height, im.width] for im in images])
+            annotations = self.feature_extractor.post_process(outputs, target_sizes)
```

Comment:
The `post_process` method is specifically defined for DETR, not sure if other models need it. Perhaps we can make sure that each object detection model's feature extractor implements a `post_process` method, that converts the model outputs to the API we are expecting.

Reply:
I'd vote for 
> Perhaps we can make sure that each object detection model's feature extractor implements a post_process method, that converts the model outputs to the API we are expecting (which is the COCO detection API).

I'd like to get @Narsil suggestion

Reply:
@Narsil can you provide your opinion here?

Reply:
i think this sounds like a perfectly fine solution

Reply:
I am not sure what my opinion on do you want ? On the output type, or the code ?

The code looks fine. You are introducing a necessity for every forward model to actually implement this `post_process` method. It really depends how much this code is model dependant.

In general, it's better to have generic code that works for every model, but it's not always possible (like `translation` or `conversational` where each models does it differently, in that case it's perfectly fine to delegate responsability to `tokenizer`/`feature_extractor`/`model` through a method.

Sorry about the very late review, notifications sometimes behaving for me, don't hesitate to ping on slack for these.

Reply:
Also pinging @sgugger which might have an opinion regarding the `post_process` method to convert from prediction to understandable result.

I think having a `post_process` akin to the tokenizer's `decode` method is a good way to ensure that the output of a model/feature_extractor combo always remains understandable, so that approach is fine for me.

Reply:
I also agree on requiring a `post_process` method for all object detection feature extractors.